### DISCUSSION TITLE
Request: Add Pause Button

### DISCUSSION THREAD

**Discussion by K6OK** [w7sst/MorseRunner#127] [Ideas]
Sometimes while practicing the XYL will call me or the phone rings.  It'd be nice to have a Pause button to freeze the program and stop the clock until I get back.

For the UI suggest following the usual standard symbols of Play - Pause - Stop  (⏵⏸⏹).   

Thanks to all for the work invested here. This program is coming along nicely!

**Comment by w7sst** [comment]
@Yagi-Berra 

Thank you for the suggestion and input - we really appreciate it. I have thought the same thing myself and agree with you. This would be really useful. In fact, I have a code snippet sitting around that partially implements a pause behavior. @scotthibbs has been tracking a separate discussion on redesign of the Run button. We'll be sure to include this input.

Thanks again and 73, Mike

**Comment by scotthibbs** [comment]
I'm not opposed to a pause button. I'm for it. 

However, in the same vain of not changing the program with Farnsworth, I think we should maintain a pure contesting mode so that all results are on the same playing field. It would be easy to use the pause button to artificially create a Farnsworth scenario if it is paused after every call heard. Or allow someone to record the CW, pause the program to replay it and "cheat" their score. Thus, I'm wanting the time limit to maintain sameness and comparability of results. 

I propose a new mode called PRACTICE instead of COMPETITION (much in the same way there is pile up and single call mode.) In this practice mode we can allow a pause and/or Farnsworth or any other aid and still maintain the competitions like in the original program.

**Reply by K6OK** [reply]
I'm not sure online speed posting of scores for competition purposes is worthwhile.  It has not gone well for other programs or sites like RufzXP or LCWO.  Cheating has been a problem.  Recommend keeping the primary focus on making MorseRunner the best contest simulator it can be.

**Comment by w7sst** [comment]
In the code, there are many places where features are disable when running
in HST mode. We would do the same thing with a Pause button - it would be
disabled when running in HST or contesting modes where scores are gathered
and later posted for comparisons with others.

**Comment by tekenny** [comment]
I think pause is good, I'd use it but also think if you pause it means you can't use that "session" for the scoreboard. For a real score you can't take any breaks!

**Reply by w7sst** [reply]
This is a good point. We should be sure that if `Pause` is used, it invalidates any session/run that is being used for the scoreboard. Also, during these so-called Competitions, we would want to disable the `Pause` button.
